File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.js
function solution(sides) { //가장 긴 변의 길이 < a+b
    let answer = 0;
    
    sides.sort((a,b)=>a-b);
    console.log(sides);
    let sum = sides[0]+sides[1];
        
    //가장 긴 변이 sides[1]
    //sides[1] - sides[0] +1 부터 sides[1] 까지
    answer = (sides[1] - (sides[1]-sides[0]))+(sum-1 - sides[1]);
    
    
    
    return answer;
}

File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.py
def solution(sides):
    answer = 0
    #1. 답이 가장 긴 변일 경우
    tri = sides[0] + sides[1]
    for i in range(max(sides), tri):
        answer +=1
    #2. 둘 중 가장 큰 변이 긴 변일 경우
    for i in range(max(sides)):
        if min(sides)+i > max(sides):
            answer +=1
    return answer